src/OrleansRuntime/Silo/SiloHost.cs

```diff
@@ -540,6 +540,12 @@ protected virtual void Dispose(bool disposing)
                         startupEvent = null;
                     }
                     this.IsStarted = false;
+
+                    if (orleans != null && orleans.DependencyResolver != null)
```

Comment:
DependencyResolver.Dispose is probably more appropriate to put inside silo `FastKill`. That way it will be called by all hosts.